web/components/navbar/index.js

```diff
@@ -23,43 +20,93 @@ const getButton = (buttonType) => {
   }
 };
 
-const Navbar = ({ displayType, buttonType }) => {
-  switch (displayType) {
-    case DISPLAY_TYPE.ONLY_LOGO:
-      return (
-        <nav>
-          <div>
-            <div>{getLogo()}</div>
-          </div>
-        </nav>
-      );
-    case DISPLAY_TYPE.LOGO_AND_BUTTON:
-      return (
-        <nav>
-          <div>
-            <div>{getLogo()}</div>
-          </div>
-          <div>
-            <div>{getButton(buttonType)}</div>
-          </div>
-        </nav>
-      );
-    default:
-      return (
-        <nav>
-          <div>
-            <div>{getLogo()}</div>
+class Navbar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hidden: false,
+      transparent: true,
+      lastY: 0,
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('scroll', this.handleScroll, { passive: true });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleScroll);
+  }
+
+  handleScroll = () => {
+    // Calculate scroll direction
+    const { lastY } = this.state;
+    const scrollingDown = (window.scrollY - lastY) >= 0;
+
+    // Calculate position
+    const offset = window.pageYOffset || document.documentElement.scrollTop;
+    const atTop = offset < 96;
+
+    // Calculate transparency
+    const transparent = offset < 256;
+
+    // Update state
+    this.setState({
+      transparent,
+      hidden: (!atTop && scrollingDown),
+      lastY: window.scrollY,
+    });
+  }
+
+  render() {
+    const { hidden, transparent } = this.state;
+    const { displayType, buttonType } = this.props;
+    const button = getButton(buttonType);
+    const links = [
```

Comment:
cleaann but can we name this something like `linksAndButton`? it doesn't just have links

Reply:
True!